railties/test/application/configuration_test.rb

```diff
@@ -2943,21 +2943,13 @@ def index
 
       app "development"
 
-      signed_id_verifier = ActiveRecord.message_verifiers["active_record/signed_id"]
+      signed_id_verifier = Rails.application.message_verifiers["active_record/signed_id"]
 
-      # generate with legacy options
-      signed_account_id_one = signed_id_verifier.generate(1, purpose: "account")
-      expected_signed_account_id_one =
-        ActiveSupport::MessageVerifier.new(app.key_generator("secret").generate_key("active_record/signed_id"), digest: "SHA256", serializer: JSON, url_safe: true)
-          .generate(1, purpose: "account")
+      secret = app.key_generator.generate_key("active_record/signed_id")
+      legacy_verifier = ActiveSupport::MessageVerifier.new(secret, digest: "SHA256", serializer: JSON, url_safe: true)
 
-      assert_equal expected_signed_account_id_one, signed_account_id_one
-
-      # verify with new options
-      signed_account_id_one_with_new_options =
-        ActiveSupport::MessageVerifier.new(app.key_generator("secret").generate_key("active_record/signed_id"))
-          .generate(1, purpose: "account")
-      assert signed_id_verifier.verify(signed_account_id_one_with_new_options, purpose: "account")
+      assert_equal "message", legacy_verifier.verify(signed_id_verifier.generate("message"))
+      assert_equal "message", signed_id_verifier.verify(legacy_verifier.generate("message"))
```

Comment:
This is much cleaner. I chose to compare the signed ID itself because [this](https://github.com/rails/rails/commit/f643919993f0de126f2d99604972bb0ba58a6296) change allows both URL-safe and non-URL-safe versions. But I agree with your implementation and we could ignore it.